src/main/java/com/codingnomads/betty/logic/services/ProbabilityToWinService.java

```diff
@@ -0,0 +1,44 @@
+package com.codingnomads.betty.logic.services;
+
+import com.codingnomads.betty.logic.models.TeamProbabilityToWin;
+import com.codingnomads.betty.logic.models.TeamSentimentScore;
+import org.springframework.stereotype.Service;
+
+@Service
+public class ProbabilityToWinService {
+
+    public TeamProbabilityToWin getProbabilityToWinFromSentimentAnalysis(TeamSentimentScore teamSentimentScore) {
+        if (teamSentimentScore.getScore() == null) {
+            throw new NullPointerException("Team Score is null");
+        }
+
+        return createTeamProbability(teamSentimentScore);
+    }
+
+    private TeamProbabilityToWin createTeamProbability(TeamSentimentScore teamSentimentScore) {
+
+        Integer teamScore = teamSentimentScore.getScore();
+
+        TeamProbabilityToWin teamProbabilityToWin = new TeamProbabilityToWin();
+        teamProbabilityToWin.setTeamName(teamSentimentScore.getTeamName());
+        teamProbabilityToWin.setMatchDateTime(teamSentimentScore.getMatchDateTime());
+        teamProbabilityToWin.setPlayingHome(teamSentimentScore.getPlayingHome());
+
+        return calculateTeamProbabilityToWin(teamScore, teamProbabilityToWin);
+    }
+
+    private TeamProbabilityToWin calculateTeamProbabilityToWin(Integer teamScore, TeamProbabilityToWin teamProbabilityToWin) {
+        if (teamScore <= 0) {
+            return calculateTeamProbabilityWithNegativeScore(teamScore, teamProbabilityToWin);
+        }
+        teamProbabilityToWin.setProbabilityToWin((double) (teamScore/100));
+        return teamProbabilityToWin;
+    }
+
+    private TeamProbabilityToWin calculateTeamProbabilityWithNegativeScore(Integer teamScore, TeamProbabilityToWin teamProbabilityToWin) {
```

Comment:
is this needed? the sentiment analysis is going to return a score in the range 0 - 100 so no negative results.

Reply:
Hey man! thank you! Not sure how you guys were planning to output the sentiment score since, depending on how you do it, you could produce negative values. If we are 100% sure we won't get negative values, then I can just throw a Runtime exception if we do get them and I would modify these two methods accordingly. Do you agree @pbxed ?

Reply:
As well, wouldnt be good if we have negative scores and you just normalize them to 0 , if there are negative scores is because the scale max-min is bigger than 0-100 so you would have to normalize everything.
The safest is if the results input come as a 0-100 as jordan suggests.

Reply:
Thank you both! Method has been removed and a custom exception is now thrown. No normalization to 0 being done